### ISSUE TITLE
`swiftlint --fix` ignores disable comment for `redundant_nil_coalescing`

### ISSUE THREAD

**Issue by windaishi** [realm/SwiftLint#6465]
## Summary

When running `swiftlint --fix`, the `redundant_nil_coalescing` rule removes both the allegedly redundant `?? nil` code AND the associated `// swiftlint:disable:this redundant_nil_coalescing` comment. This is problematic because:

1. The disable comment should prevent any auto-fix from being applied
3. The auto-fix silently removes the developer's explicit disable comment

## Environment

- **SwiftLint version**: 0.63.2
- **Installation method**: mise
- **Xcode version**: Xcode 26.2 (Build version 17C52)
- **Swift version**: Apple Swift version 6.2.3 (swiftlang-6.2.3.3.21 clang-1700.6.3.2)
- **macOS version**: 26.2

## Steps to Reproduce

1. Create a Swift file with the following content:

```swift
let result: SomeType? = try someOptionalOperation().flatMap { value in
    if someCondition {
        return try processValue(value)
    }
    return nil
} ?? nil // swiftlint:disable:this redundant_nil_coalescing - false positive
```

2. Run `swiftlint --fix`

## Expected Behavior

The line should remain unchanged because the `redundant_nil_coalescing` rule is explicitly disabled for this line via the `// swiftlint:disable:this redundant_nil_coalescing` comment.

The code should stay as:
```swift
} ?? nil // swiftlint:disable:this redundant_nil_coalescing - false positive
```

## Actual Behavior

SwiftLint removes both the `?? nil` AND the disable comment, changing the code to:

```swift
}
```

This is a double violation:
1. The auto-fix was applied despite the disable comment
2. The disable comment itself was removed from the codebase
